src/components/Dates/components/ExpMonthInput.tsx

```diff
@@ -1,16 +1,15 @@
-import { ChangeEvent, Dispatch } from "react"
-import { FormActions } from "../../../reducers/form-reducer"
-import { formatMonth } from "../../../helpers/helpers"
+import { ChangeEvent } from "react"
 import { EFormaActions, EFormData } from "../../../enums/Form"
+import { formatMonth } from "../../../helpers/helpers"
+import { useForm } from "../../../hooks/useForm"
 
-type ExpMonthInputProps = {
-    state: string
-    dispatch: Dispatch<FormActions>
-    expMonthError?: string;
-    expYearError?: string;
-}
+const ExpMonthInput = () => {
 
-const ExpMonthInput = ({ state, dispatch, expMonthError, expYearError }: ExpMonthInputProps) => {
+    const { state, dispatch, errors } = useForm()
+    const { expMonth } = state
+    const expMonthError = errors[EFormData.ExpMonth];
+    const expYearError = errors[EFormData.ExpYear];
+    
 
     const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
```

Comment:
this handleChange should be ideal to have it in the context so you remove most of the logic to the context and keep clean the component